chain/chain/src/types.rs

```diff
@@ -407,9 +413,38 @@ mod tests {
             2,
             b1.header.inner.epoch_id.clone(),
             approvals,
-            signer.clone(),
+            &*signer,
         );
         assert!(signer.verify(b2.hash().as_ref(), &b2.header.signature));
         assert_eq!(b2.header.inner.total_weight.to_num(), 3);
     }
 }
+
+/// Verifies that chunk's proofs in the header match the body.
+pub fn validate_chunk_proofs(
```

Comment:
I added same function to RuntimeAdapter, it's up to you to choose better place.

Reply:
yeah, I removed it from there. I think we should keep RuntimeAdapter as small as possible.